### ISSUE TITLE
cifar10 CNN训练问题

### ISSUE THREAD

**Issue by supengyu** [dmlc/cxxnet#244]
在cxxnet.exe程序训练cifar10图片库是遇到问题，程序异常退出，如下所示：

E:\deepLearning\cxxnet-master\bin>E:\deepLearning\cxxnet-master\bin\cxxnet.exe E
:\deepLearning\cxxnet-master\bin\ImageNet\ImageNet.conf
Use CUDA Device 0: GeForce GTX 980 Ti
finish initialization with 1 devices
Initializing layer: conv1
Initializing layer: relu1
Initializing layer: pool1
Initializing layer: lrn1
Initializing layer: conv2
Initializing layer: relu2
Initializing layer: pool2
Initializing layer: lrn2
Initializing layer: conv3
Initializing layer: relu3
Initializing layer: pool3
Initializing layer: flatten1
Initializing layer: fc4
Initializing layer: softmax1
LossLayer is an self-loop Layer代码跟踪发现：在LoadModel函数中执行this->FreeSpace()时意外退出了，不知道是什么问题，

配置文件imageNet.conf如下：
# Configuration for ImageNet
# Acknowledgement:
# Ref: http://www.cs.toronto.edu/~fritz/absps/imagenet.pdf
# The scheduling parameters is adapted from Caffe(http://caffe.berkeleyvision.org/)

data = train
iter = imgrec
# image_list = "../../NameList.train"

  image_rec  = "E:/deepLearning/cxxnet-master/bin/data/cifar10train.bin"
  image_root = "E:/deepLearning/cifar-10/cifar-10-py-colmajor/batch_1/"
  image_mean = "E:/deepLearning/cxxnet-master/bin/models/image_net_mean.bin"
  rand_crop=1
  rand_mirror=1
iter = threadbuffer
iter = end

eval = test
iter = imgrec
# image_list = "../../NameList.test"

  image_rec = "E:/deepLearning/cxxnet-master/bin/data/cifar10test.bin"
  image_root = "E:/deepLearning/cifar-10/cifar-10-py-colmajor/test_batch/"
  image_mean = "E:/deepLearning/cxxnet-master/bin/models/image_net_mean.bin"
# no random crop and mirror in test

iter = end

netconfig=start
layer[0->1] = conv:conv1
  kernel_size = 5
  stride = 1
  nchannel = 32
layer[1->2] = relu:relu1
layer[2->3] = max_pooling:pool1
  kernel_size = 3
  stride = 2
layer[3->4] = lrn:lrn1
  local_size = 5
  alpha = 0.0001
  beta = 0.75
  knorm = 1
###### 

layer[4->5] = conv:conv2
  ngroup = 1
  nchannel = 32
  kernel_size = 5
  pad = 1
layer[5->6] = relu:relu2
layer[6->7] = max_pooling:pool2
  kernel_size = 3
  stride = 2
layer[7->8] = lrn:lrn2
  local_size = 5
  alpha = 0.0001
  beta = 0.75
  knorm = 1
###### 

layer[8->9] = conv:conv3
  nchannel = 64
  kernel_size = 3
  pad = 1
layer[9->10]= relu:relu3
layer[10->11] = max_pooling:pool3
 kernel_size = 3
  stride = 2
layer[11->12] = flatten:flatten1
layer[12->13] = fullc:fc4
  nhidden = 512
  init_sigma = 0.005
  init_bias = 1.0
layer[13->14] = softmax:softmax1

netconfig=end
# evaluation metric

metric = error
metric = rec@1
metric = rec@5

max_round = 45
num_round = 45
# input shape not including batch

input_shape = 3,32,32

batch_size = 128
# global parameters in any sectiion outside netconfig, and iter

momentum = 0.9
wmat:lr  = 0.01
wmat:wd  = 0.0005

bias:wd  = 0.000
bias:lr  = 0.02
# all the learning rate schedule starts with lr

lr:schedule = expdecay
lr:gamma = 0.1
lr:step = 100000

save_model=1
model_dir=models
# random config

random_type = xavier
# new line

**Comment by piiswrong** [comment]
Development on CXXNet has stopped. We recommend that you move to MXNet

**Comment by tiankong1993** [comment]
在吗 可否留个qq 我想请教一下caffe的问题
